File: 2224. Minimum Number of Operations to Convert Time/solution.cpp
class Solution {
public:
  int convertTime(string current, string correct) {
    int currentHours = stoi(current.substr(0, 2));
    int currentMinutes = stoi(current.substr(3, 2));
    int correctHours = stoi(correct.substr(0, 2));
    int correctMinutes = stoi(correct.substr(3, 2));

    int count = correctHours - currentHours;
    int difference = 0;
    if (currentMinutes < correctMinutes){
      difference = correctMinutes - currentMinutes;
    }
    else if (currentMinutes > correctMinutes){
      count -= 1;
      difference = 60 - currentMinutes + correctMinutes;
    }

    if (floor(difference / 15) > 0){
      int amount = floor(difference / 15);
      count += amount;
      difference -= amount * 15;
    }
    if (floor(difference / 5) > 0){
      int amount = floor(difference / 5);
      count += amount;
      difference -= amount * 5;
    }
    if (floor(difference / 1) > 0){
      int amount = floor(difference / 1);
      count += amount;
      difference -= amount * 1;
    }

    return count;
  }
};

File: 2224. Minimum Number of Operations to Convert Time/solution.py
class Solution(object):
  def convertTime(self, current, correct):
    """
    :type current: str
    :type correct: str
    :rtype: int
    """
    currentHours = int(current[0:2])
    currentMinutes = int(current[3:5])
    correctHours = int(correct[0:2])
    correctMinutes = int(correct[3:5])
    
    count = correctHours - currentHours
    difference = 0
    if currentMinutes < correctMinutes:
      difference = correctMinutes - currentMinutes
    elif currentMinutes > correctMinutes:
      count -= 1
      difference = 60 - currentMinutes + correctMinutes
    
    if difference / 15 > 0:
      amount = difference / 15
      count += amount
      difference -= amount * 15
    if difference / 5 > 0:
      amount = difference / 5
      count += amount
      difference -= amount * 5
    if difference / 1 > 0:
      amount = difference / 1
      count += amount
      difference -= amount * 1
      
    return count